File: core/src/com/xenoage/zong/core/position/MPElement.java
package com.xenoage.zong.core.position;

import com.xenoage.zong.core.music.MusicElement;


/**
 * Interface for all musical elements, for which an {@link MP} can be retrieved.
 * 
 * @author Andreas Wenger
 */
public interface MPElement
	extends MusicElement {

	/**
	 * Gets the parent element, or null.
	 */
	public MPContainer getParent();

}

File: core/src/com/xenoage/zong/core/position/MPElement.kt
package com.xenoage.zong.core.position

import com.xenoage.zong.core.music.MusicElement


/**
 * Interface for all musical elements, for which an [MP] can be retrieved.
 */
interface MPElement : MusicElement {

	/** The parent element, or null if not part of a score. */
	var parent: MPContainer?

	/**
	 * Gets the [MP] of this element by querying its parent, or null if it is unknown.
	 */
	val mp: MP?
		get() = this.parent?.getChildMP(this)

}
